src/services/tier/useTier.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+
+export const TierNames = {
+  BASIC: 'basic',
+  LITE: 'lite',
+  PRO: 'pro',
+  ENTERPRISE: 'enterprise',
+} as const
+
+export const TierSchema = z
+  .object({
+    tierName: z.nativeEnum(TierNames),
+  })
+  .nullish()
+
+export interface UseTierArgs {
+  provider: string
+  owner: string
+}
+
+const query = `
+  query OwnerTier($owner: String!) {
+    owner(username:$owner){
+      plan {
+        tierName
+      }
+    }
+  }
+`
+
+export const useTier = ({ provider, owner }: UseTierArgs) =>
+  useQuery({
+    queryKey: ['OwnerTier', provider, owner, query],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          owner,
+        },
+      }).then((res) => {
+        return TierSchema.parse(res?.data?.owner?.plan) ?? {}
```

Comment:
Can you switch this to a safe parse and check to see if the request was successful and if not return a rejected promise so that we can handle that error a little more gracefully, as right now it won't come up as a network error but as a regular JS error that crashed the app.

Reply:
Sureee! When should we use parse vs safeParse? I suppose we prefer on safeParse and handle promises right?

Reply:
Should always use safeParse ... sorry for my previous indiscretions. It's much better/proper to handle errors rather then throwing them all the time. The reason for `Promise.reject` is because we have lint rules that you can't throw objects, so we use `Promise.reject` in it's place as Tanstack accepts that in the same manner as throwing an error, then will throw it up into the proper ErrorBoundary

Reply:
Makes sense, I assumed we transitioned into safeParse as we discovered more of zod 👌